docs/topics/compatibility-guide-18.md

```diff
@@ -0,0 +1,623 @@
+
+- _source_: source-incompatible change stops code that used to compile fine (without errors or warnings) from compiling
+  anymore
+- _binary_: two binary artifacts are said to be binary-compatible if interchanging them doesn't lead to loading or
+  linkage errors
+- _behavioral_: a change is said to be behavioral-incompatible if the same program demonstrates different behavior
+  before and after applying the change
+
+Remember that those definitions are given only for pure Kotlin. Compatibility of Kotlin code from the other languages
+perspective
+(for example, from Java) is out of the scope of this document.
+
+## Language
+
+<!--
+### Title
+
+> **Issue**: [KT-NNNNN](https://youtrack.jetbrains.com/issue/KT-NNNNN)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**:
+>
+> **Deprecation cycle**:
+>
+> - 1.6.20: report a warning
+> - 1.8.0: raise the warning to an error
+-->
+
+
+### Prohibit the delegation of super calls to an abstract superclass member
+
+> **Issues**: [KT-45508](https://youtrack.jetbrains.com/issue/KT-45508), [KT-49017](https://youtrack.jetbrains.com/issue/KT-49017), [KT-38078](https://youtrack.jetbrains.com/issue/KT-38078)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+> 
+> **Short summary**: Kotlin will report a compile error when an explicit or implicit super call is delegated 
+> to an _abstract_ member of the superclass, even if there's a default implementation in a super interface
+>
+> **Deprecation cycle**:
+>
+> - 1.5.20: introduce a warning when non-abstract classes that do not override all abstract members are used
+> - 1.7.0: report a warning if a super call, in fact, accesses an abstract member from a superclass
+> - 1.7.0: report an error in all affected cases if the `-Xjvm-default=all` or `-Xjvm-default=all-compatibility` compatibility modes are enabled;
+>   report an error in the progressive mode
+> - 1.8.0: report an error in cases of declaring a concrete class with a non-overridden abstract method from the superclass, and 
+>   super calls of methods of `Any` overridden as abstract in the superclass
+> - 1.9.0: report an error in all affected cases, including explicit super calls to an abstract method from the super class
+
+
+### Deprecate confusing grammar in `when-with-subject`
+
+> **Issue**: [KT-48385](https://youtrack.jetbrains.com/issue/KT-48385)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin 1.6 deprecated several confusing grammar constructs in `when` condition expressions
+>
+> **Deprecation cycle**:
+>
+> - 1.6.20: introduce a deprecation warning on the affected expressions
+> - 1.8.0: raise this warning to an error,
+>   `-XXLanguage:-ProhibitConfusingSyntaxInWhenBranches` can be used to temporarily revert to the pre-1.8 behavior.
+> - \>= 1.9: repurpose some deprecated constructs for new language features
```

Comment:
```suggestion
> - \>= 1.9.0: repurpose some deprecated constructs for new language features
```

Reply:
For `>= 1.x` I intentionally don't specify the last version part, to emphasize that it's approximate

Reply:
Ok, thanks!